disnake/ui/action_row.py

```diff
@@ -267,19 +356,100 @@ def add_text_input(
             )
         )
 
+    @property
+    def _underlying(self) -> ActionRowComponent:
```

Comment:
Atm, it should create a new `ActionRowComponent` every time `_underlying` is accessed, so this shouldn't specifically be an issue. I thought this would make more sense, as it would properly support modifying components in-place.

The pre-existing system would be more lightweight as it did not create a new `ActionRowComponent` each time `_underlying` was accessed, but it would be "desynced" when a component was modifed.

Reply:
Hmm, how would it be desynced? The ui types all have setters to modify their respective `_underlying` fields

Reply:
Hmm, how would it be desynced? The ui types all have setters to modify their respective `_underlying` fields

Reply:
You're absolutely correct, my mistake. We can decide on which approach is preferred, and I'll revert if need be. Ultimately, I don't think `ActionRow._underlying` will be accessed that frequently, so either way is fine with me.

Reply:
How was this resolved?

Reply:
This was left as-is, as the performance hit seems negligible, considering the fact that `_underlying` will only really be accessed upon sending of the action row.

Reply:
Gotcha.